src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -1077,3 +1068,43 @@ export const selectEquityTotalBalance = createDeepEqualOutputSelector(
     )
   },
 )
+
+export const selectPortfolioTotalBalanceUserCurrencyByChainId = createDeepEqualOutputSelector(
+  selectPortfolioUserCurrencyBalancesByAccountId,
+  portfolioUserCurrencyBalancesByAccountId => {
+    return Object.entries(portfolioUserCurrencyBalancesByAccountId).reduce(
+      (acc, [accountId, accountBalancesUserCurrencyByAssetId]) => {
+        const { chainId } = fromAccountId(accountId)
+        const totalBalanceUserCurrency = Object.values(accountBalancesUserCurrencyByAssetId).reduce(
+          (accountTotal, assetBalanceUserCurrency) =>
+            accountTotal.plus(bnOrZero(assetBalanceUserCurrency)),
+          bn(0),
+        )
+
+        acc[chainId] = bnOrZero(acc[chainId]).plus(totalBalanceUserCurrency).toString()
+        return acc
+      },
+      {} as Record<ChainId, string>,
+    )
+  },
+)
+
+export const selectWalletConnectedChainIdsSorted = createDeepEqualOutputSelector(
+  selectPortfolioTotalBalanceUserCurrencyByChainId,
+  portfolioTotalBalanceUserCurrencyByChainId => {
+    const chainAdapterManager = getChainAdapterManager()
+    return orderBy(
+      Object.entries(portfolioTotalBalanceUserCurrencyByChainId).map(
+        ([chainId, totalBalanceUserCurrency]) => {
+          return {
+            chainId,
+            totalBalanceUserCurrency: Number(totalBalanceUserCurrency),
```

Comment:
It might be safer to `bnOrZero()` here.

Reply:
This guy's only used for sorting of fiat amounts, so precision isn't a concern here. The max value of javascript double should be sufficient to handle any mega fiat balance any user could have so we should be okay?